mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageManager.java

```diff
@@ -734,42 +733,42 @@ private MoveResult move(Iterator<MailboxMessage> originalRows, MailboxSession se
         SortedMap<MessageUid, MessageMetaData> copiedUids = collectMetadata(to.copy(originalRows, session));
 
         ImmutableMap.Builder<MessageUid, MailboxMessage> messagesMap = ImmutableMap.builder();
-        for (MailboxMessage message: originalRows.getEntriesSeen()) {
+        for (MailboxMessage message : originalRows.getEntriesSeen()) {
             messagesMap.put(message.getUid(), immutableMailboxMessageFactory.from(to.getMailboxEntity().getMailboxId(), message));
         }
         dispatcher.added(session, copiedUids, to.getMailboxEntity(), messagesMap.build());
-        dispatcher.moved(session,
-            MessageMoves.builder()
-                .previousMailboxIds(getMailboxEntity().getMailboxId())
-                .targetMailboxIds(to.getMailboxEntity().getMailboxId(), getMailboxEntity().getMailboxId())
-                .build(),
-            messagesMap.build());
-
+        if (!messagesMap.build().isEmpty()) {
```

Comment:
sorry if I have been unclear but I think that it's not the responsibility of move method to do that filtering.
We should be able to build the MessageMoves message with an empty list and the dispatcher should do the filtering.
We could, for example, add a "isNoop" on MessageMoves (and other evenets) to help filtering events.

Reply:
Sure, that would be less invasive.

If no-one do it before, I will do it on Thursday.